tracecat/agent/adapter/vercel.py

```diff
@@ -760,12 +765,225 @@ def collect_current_part_end_events(
         return events
 
     async def handle_event(
```

Comment:
maybe we could split this up into subclasses instead of using conditionals everywhere here